OpenSim/Common/APDMDataReader.cpp

```diff
@@ -213,36 +213,36 @@ APDMDataReader::extendRead(const std::string& fileName) const {
     return tables;
 }
 
-void APDMDataReader::find_start_column(std::vector<std::string> tokens, 
-    std::vector<std::string> search_labels,
-    const std::string& sensorName,
-    std::vector<int>& indices, bool newFromat) const {
+void APDMDataReader::find_start_column(std::vector<std::string> tokens,
+                                       std::vector<std::string> search_labels, const std::string& sensorName,
+                                       std::vector<int>& indices, bool newFormat) const {
     // Search for "sensorName/{search_labels} in tokens, append result to indices if found"
     std::string firstLabel = sensorName + search_labels[0];
     // look for first label, when found check/confirm the rest. Out of order is not supported
     int found_index = -1;
-    std::vector<std::string>::iterator it = std::find(tokens.begin(), tokens.end(), firstLabel);
+    std::vector<std::string>::iterator it =
+            std::find(tokens.begin(), tokens.end(), firstLabel);
     if (it != tokens.end()) {
         found_index = static_cast<int>(std::distance(tokens.begin(), it));
-        // now check the following indices for match with remaining search_labels 
+        // now check the following indices for match with remaining search_labels
         bool match = true;
-        for (int remaining = 1; remaining < search_labels.size() && match; remaining++) {
-            match = tokens[found_index + remaining].
-                compare(sensorName + search_labels[remaining]) == 0;
+        for (int remaining = 1; remaining < (int)search_labels.size() && match;
+             remaining++) {
+            match = tokens[found_index + remaining].compare(
+                    sensorName + search_labels[remaining]) == 0;
         }
         if (match) {
-            if (newFromat)
-                indices.push_back(found_index - 3); // Three extra  comma separated fields in header before imu name
-            else
+            if (newFormat) {
+                // Three extra comma separated fields in header before imu name
+                indices.push_back(found_index - 3);
+            } else {
                 indices.push_back(found_index);
-                return;
-            }
-            else { // first label found but the remaining didn't. Throw
-                throw Exception{ "Expected labels for sensor " + sensorName + 
-                    " were not found." };
             }
+        } else { // first label found but the remaining didn't. Throw
+            throw Exception{"Expected labels for sensor " + sensorName +
+                            " were not found."};
         }
-    return; // not found
     }
 }
```

Comment:
@aymanhab can you look over these changes carefully?

Reply:
@chrisdembia I will but the PR has 87 files all over the place, can we piecemeal it in a meaningful fashion? I also worry that being on the bleeding edge of a specific compiler/version may result in compilation being less robust for other environments (users or ci with slightly different version/upgrade). We can make sure the code produces no warnings to test but not build it into CMakeLists that are distributed or live on master. Thoughts?

Reply:
My comment here was regarding this specific file; the majority of the other changes are much simpler.

Would you be okay with piecemealing through Reviewable? That allows you or other reviewers to review a few files at a time without needing to split the PR.

My preference is to keep the settings I put on this branch: generate more warnings, and present these warnings as errors. This promotes the best hygiene for our codebase, and makes our codebase more robust to the latest compilers.

On the master branch, perhaps we can leave `-Wextra -Wall` but remove `-Werror`.

Reply:
Would you be willing to try leaving `-Werror` on the master branch for a while, and remove it if it turns out to cause headaches?

Reply:
Thanks @chrisdembia, will use Reviewable as you suggested to split the review.
Again I feel warnings as errors is more drastic considering the possible variety of platforms, compilers/upgrades etc. We wouldn't know if it's mere headache or a nightmare for users until release, so definitely release/tag without -Werror but we can leave it for a bit on master to test against ci. We'll never know who else is using it and need to remember to revert the flag before public release.

Reply:
Thanks @aymanhab. I've removed `-Werror` for now.